File: WebClient SDK/src/default/files/clientKrPermissionsGetFileVersionsExtension.ts
import { CardGetFileVersionsExtension, ICardGetFileVersionsExtensionContext } from 'tessa/cards/extensions';
import { UIContext } from 'tessa/ui';
import { ICardModel } from 'tessa/ui/cards';
import { KrToken } from 'tessa/workflow';

export class ClientKrPermissionsGetFileVersionsExtension extends CardGetFileVersionsExtension {

  public beforeRequest(context: ICardGetFileVersionsExtensionContext) {
    const editor = UIContext.current.cardEditor;
    let model: ICardModel;
    let token: KrToken;

    if (editor
      && (model = editor.cardModel!)
      && (token = KrToken.tryGet(model.card.info)!)
    ) {
      token.setInfo(context.request.info);
    }
  }

}

File: Source/Extensions/Tessa.Extensions.Default.Client/Files/ClientKrPermissionsGetFileVersionsExtension.cs
using System.Threading.Tasks;
using Tessa.Cards.Extensions;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.UI;
using Tessa.UI.Cards;

namespace Tessa.Extensions.Default.Client.Files
{
    public sealed class ClientKrPermissionsGetFileVersionsExtension :
        CardGetFileVersionsExtension
    {
        #region Base Overrides

        public override Task BeforeRequest(ICardGetFileVersionsExtensionContext context)
        {
            ICardEditorModel editor = UIContext.Current.CardEditor;
            ICardModel model;
            KrToken token;

            if (editor != null
                && (model = editor.CardModel) != null
                && (token = KrToken.TryGet(model.Card.Info)) != null)
            {
                token.Set(context.Request.Info);
            }

            return Task.CompletedTask;
        }

        #endregion
    }
}